### PR TITLE
Add app status spike findings

### PR DESCRIPTION
**Pull request by J-Thompson12**
Closes: #1188 

**What changed?**
Added findings for the above app status spike.

**Why?**

**How did you test it?**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: doc/design/app-status.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+The UI will remain the same while the CLI will need to be updated like so to match: 
+```
+KIND              NAME                     STATUS
+Application       josh-pod-info            ready
+Deployment        backend                  ready
+ReplicaSet        backend-6b944d8b         ready
+Pod               backend-6b944d8b-tzxnl   ready
+Deployment        frontend                 ready
+Kustomization     podinfo-deploy           ready
+```
```
Comment (J-Thompson12) [inline doc/design/app-status.md:36]: I was unable to get products thoughts on this.

File: doc/design/app-status.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+***Deployment details***:
+KIND              NAME                     STATUS
+Application       josh-pod-info            ready
+Deployment        backend                  ready
+ReplicaSet        backend-6b944d8b         ready
+Pod               backend-6b944d8b-tzxnl   ready
+Deployment        frontend                 ready
+Kustomization     podinfo-deploy           ready
+```
```
Comment (J-Thompson12) [inline doc/design/app-status.md:54]: There is a discussion about how much info to show on here as well as how to show the hierarchy. Any ideas would be helpful.

File: doc/design/app-status.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
+### Status of my Application
+
+As a user, I expect to see the status of my application running across multiple clusters. When I go to my application list page I should be able to quickly see the "health" of each application in each cluster within a timely manner.
+
+## Problem to be Solved
+
+- Come up with a way for the CLI to have parity with the UI status request.
```
[the remark this answers is not in the document]
Comment (J-Thompson12) [reply doc/design/app-status.md:8]: I was just thinking this parity was in how the user would view it in the UI/CLI. I have added some info on what could be used to get the status for both.

File: doc/design/app-status.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        - "aws-marketplace:RegisterUsage"
+        Resource: '*'
+
+managedNodeGroups:
+- name: ng1
+  instanceType: m5.large
+  desiredCapacity: 3
+
+```
+It took a little less than 2 seconds to get all the info needed to generate the graph. If scaling is needed in the future kstatus has the ability to collect and save the last reported status for a resource https://github.com/kubernetes-sigs/cli-utils/blob/v0.26.1/pkg/kstatus/polling/collector/collector.go. We could save that information in memory or if need be a redis cluster. I believe that scaling may not be needed at this point and will not be hard to implement in the future using the tools we currently have.
```
[the remark this answers is not in the document]
Comment (J-Thompson12) [reply doc/design/app-status.md]: I included how flux is currently using kstatus cache and confirmed with them that it works for any size cluster.

File: doc/design/app-status.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+
+Kstatus does not work for helm. The UI currently does not support a helm application. The way the server code handles the status though should make it easy to include helm. Instead of using status.Compute to get the status for helm we could use the information returned when s.kube.list is called. GetReconciledObjects somewhat handles helm but it needs to have code added to fully support getting the status for helm.
```
[the remark this answers is not in the document]
Comment (J-Thompson12) [reply doc/design/app-status.md]: Confirmed with flux team that kstatus works for helmrelease/helmchart so the statuses would be the same.

**Comment by Callisto13** [comment]
can we close this now?

**Comment by jpellizzari** [comment]
> can we close (or merge) this now?

...yeah I think that is safe

**Pull request closed without merging**